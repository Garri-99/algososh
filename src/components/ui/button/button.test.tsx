import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "./button";

describe("Button component renders correctly", () => {
  afterEach(cleanup);

  it("with text", () => {
    render(<Button text="Button" />);
    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("Button");
    expect(button).toMatchSnapshot();
  });

  it("without text", () => {
    render(<Button />);
    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("");
    expect(button).toMatchSnapshot();
  });

  it("with disabled state", () => {
    render(<Button disabled />);
    const button = screen.getByTestId("button");
    expect(button).toBeDisabled();
    expect(button).toMatchSnapshot();
  });

  it("with loader", () => {
    render(<Button isLoader />);
    const button = screen.getByTestId("button");
    const loader = screen.getByTestId("loader");
    expect(button).toContainElement(loader);
    expect(button).toMatchSnapshot();
  });

  it("with callback", () => {
    const mockCb = jest.fn();
    render(<Button onClick={mockCb} />);
    const button = screen.getByTestId("button");
    userEvent.click(button);
    expect(mockCb).toBeCalledTimes(1);
  });
});
