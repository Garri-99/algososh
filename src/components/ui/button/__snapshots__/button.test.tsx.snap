// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with disabled state 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component renders correctly with loader 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    data-testid="loader"
    src="loader.svg"
  />
</button>
`;

exports[`Button component renders correctly with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  >
    Button
  </p>
</button>
`;

exports[`Button component renders correctly without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
